torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -165,6 +206,35 @@ class TORCH_API RpcAgent {
   // Add GIL wait time data point to metrics
   virtual void addGilWaitTime(const std::chrono::microseconds gilWaitTime) = 0;
   friend class PythonRpcHandler;
+
+  // Map that stores metadata for RPC's that may need to be re-tried as well as
+  // the timepoint at which we should re-try them.
+  std::map<steady_clock_time_point, std::shared_ptr<RpcRetryInfo>> rpcRetryMap_;
```

Comment:
This should be a map from time_point to a vector of RpcRetryInfo since we could have RpcRetryInfo for two RPCs with the same timepoint.